performance-tests/perf-tests.sh

```diff
@@ -0,0 +1,220 @@
+}
+
+# expects TEST_NAME, TOPIC, ENDPOINT, PRODUCER_RESULT and CONSUMER_RESULT to be set
+doPerfTest () {
+  doCreateTopic ${ENDPOINT} ${TOPIC}
+  warmUp ${ENDPOINT} ${TOPIC}
+
+  producerPerf ${ENDPOINT} ${TOPIC} ${NUM_RECORDS} ${PRODUCER_RESULT}
+  consumerPerf ${ENDPOINT} ${TOPIC} ${NUM_RECORDS} ${CONSUMER_RESULT}
+
+  doDeleteTopic ${ENDPOINT} ${TOPIC}
+}
+
+onExit() {
+  for cmd in "${ON_SHUTDOWN[@]}"
+  do
+    eval ${cmd}
+  done
+}
+
+trap onExit EXIT
+
+TMP=$(mktemp -d)
+ON_SHUTDOWN+=("rm -rf ${TMP}")
+
+# Bring up Kafka
+ON_SHUTDOWN+=("runDockerCompose down")
+runDockerCompose pull
+runDockerCompose up --detach --wait kafka
+
+# Warm up the broker - we do this separately as we might want a longer warm-up period
+doCreateTopic broker1:9092 warmup-topic
+warmUp broker1:9092 warmup-topic ${WARM_UP_NUM_RECORDS_POST_BROKER_START}
+doDeleteTopic broker1:9092 warmup-topic
+
+echo -e "${GREEN}Running test cases, number of records = ${NUM_RECORDS}, record size ${RECORD_SIZE}${NOCOLOR}"
+
+PRODUCER_RESULTS=()
+CONSUMER_RESULTS=()
+for t in $(find ${PERF_TESTS_DIR} -type d -regex '.*/'${TEST} | sort)
+do
+  TESTNAME=$(basename $t)
+  TEST_TMP=${TMP}/${TESTNAME}
+  mkdir -p ${TEST_TMP}
+  PRODUCER_RESULT=${TEST_TMP}/producer.json
+  CONSUMER_RESULT=${TEST_TMP}/consumer.json
+  TOPIC=perf-test-${RANDOM}
+
+  echo -e "${GREEN}Running ${TESTNAME} ${NOCOLOR}"
+
+  TESTNAME=${TESTNAME} TOPIC=${TOPIC} PRODUCER_RESULT=${PRODUCER_RESULT} CONSUMER_RESULT=${CONSUMER_RESULT} . ${t}/run.sh
+
+  PRODUCER_RESULTS+=(${PRODUCER_RESULT})
+  CONSUMER_RESULTS+=(${CONSUMER_RESULT})
+done
+
+# Summarise results
+
+echo -e "${GREEN}Producer Results ${NOCOLOR}"
+
+jq -r -s '(["Name","Sent","Rate rec/s", "Rate Mi/s", "Avg Lat ms", "Max Lat ms", "Percentile50", "Percentile95", "Percentile99", "Percentile999"] | (., map(length*"-"))),
+           (.[] | [ .name, (.values | last | .[]) ]) | @tsv' "${PRODUCER_RESULTS[@]}" | column -t -s $'\t'
+
+echo -e "${GREEN}Consumer Results ${NOCOLOR}"
+
+jq -r -s '(["Name","Consumed Mi","Consumed Mi/s", "Consumed recs", "Consumed rec/s", "Rebalance Time ms", "Fetch Time ms", "Fetch Mi/s", "Fetch rec/s"] | (., map(length*"-"))),
+           (.[] | [ .name, (.values  | last | .[]) ]) | @tsv' "${CONSUMER_RESULTS[@]}" | column -t -s $'\t'
+
+
+# Write output for integration with Kibana in the CI pipeline
+# This maintains the existing interface
+if [[ ! -z "${KIBANA_OUTPUT_DIR}" && -d "${KIBANA_OUTPUT_DIR}" ]]; then
+  for PRODUCER_RESULT in ${PRODUCER_RESULTS[@]}
+  do
+    DIR=${KIBANA_OUTPUT_DIR}/$(jq -r '.name' ${PRODUCER_RESULT})
+    mkdir -p ${DIR}
+    jq '.values | last | [{name: "AVG Latency", unit: "ms", value: .avg_lat_ms},
+                          {name: "95th Latency", unit: "ms", value: .percentile95},
+                          {name: "99th Latency", unit: "ms", value: .percentile99}]' ${PRODUCER_RESULT} > ${DIR}/producer.json
+  done
+
+  for CONSUMER_RESULT in ${CONSUMER_RESULTS[@]}
```

Comment:
shellcheck flagging this as an [error](https://www.shellcheck.net/wiki/SC2068)
```suggestion
  for CONSUMER_RESULT in "${CONSUMER_RESULTS[@]}"
```